### PR TITLE
Count docs in index when loading sample data

### PR DESCRIPTION
**Pull request by dhruvkb**
## Fixes

Addresses #2359 by @sarayourfriend 

## Description

This PR adds another check to the load-sample-data script to check that the new indices created have 5000 sample documents in them.

## Checklist

- [x] My pull request has a descriptive title (not a vague title like`Update index.md`).
- [x] My pull request targets the _default_ branch of the repository (`main`) or a parent feature branch.
- [x] My commit messages follow [best practices][best_practices].
- [x] My code follows the established code style of the repository.
- [x] I added or updated tests for the changes I made (if applicable).
- [ ] I added or updated documentation (if applicable).
- [x] I tried running the project locally and verified that there are no visible errors.
- [ ] I ran the DAG documentation generator (if applicable).

[best_practices]:
  https://git-scm.com/book/en/v2/Distributed-Git-Contributing-to-a-Project#_commit_guidelines

## Developer Certificate of Origin

<details>
<summary>Developer Certificate of Origin</summary>

```
Developer Certificate of Origin
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.
1 Letterman Drive
Suite D4700
San Francisco, CA, 94129

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.


Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.
```

</details>

### CODE REVIEW

File: docker/es/justfile [hunk 1]
```diff
@@ -42,3 +43,22 @@ NO_COLOR := "\\033[0m"
     just ../../_loop \
     '"$(just docker/es/check-index {{ index }} {{ dest }})" != "1"' \
     "Waiting for index '{{ index }}' to be ready..."
+
+# Check that the given index has the given number of docs
+@check-count index="image" count="5000":
```
Comment (AetherUnbound) [inline docker/es/justfile:48]: This looks quite similar to the above `wait-for-index` commands - do you think it makes sense to consolidate some logic?

File: docker/es/justfile [hunk 2]
```diff
@@ -42,3 +43,22 @@ NO_COLOR := "\\033[0m"
...
+    | wc -l \
+    | xargs
+
+# Wait for the index to have the given number of docs
+@wait-for-count index="image" count="5000":
+    # The just command on the second line is executed in the context of the
+    # root directory and so must be prefixed with `docker/es/`.
+    just ../../_loop \
+    '"$(just docker/es/check-count {{ index }} {{ count }})" != "1"' \
+    "Waiting for index '{{ index }}' to have {{ count }} docs..."
```
Comment (AetherUnbound) [inline docker/es/justfile:64]: Oddly enough, the `Waiting for index <> to be ready` lines get logged, but these `Waiting for index <> to have <> docs` lines do not... 🤔 do you know why that is? They both look like they're invoked the same.

**Review by AetherUnbound — APPROVED** [review]
Tested locally and it worked as expected!

File: docker/es/justfile (hunk 2 above)
Comment (dhruvkb) [reply docker/es/justfile:64]: Could be that it passed immediately? The line will not be printed if the condition passes, it's only printed when waiting to retry.

Comment (sarayourfriend) [inline docker/es/justfile:64]:
```suggestion
    "Waiting for index '{{ index }}' to have {{ count }} docs..." \
    "10s"
```

**Review by sarayourfriend — APPROVED** [review]
Seems like another good addition. The default timeout for the `_loop` recipe is 5 minutes, it seems like for this instance it could be way lower, right? Maybe even like 10 seconds, I wouldn't expect it to take more than that before I thought something was seriously going wrong locally.

I left a suggested change that seems to work locally to add a 10 second timeout. Either way this looks good.

File: docker/es/justfile (hunk 2 above)
Comment (AetherUnbound) [reply docker/es/justfile:64]: Ahh, that's got to be it!

File: docker/es/justfile (hunk 1 above)
Comment (dhruvkb) [reply docker/es/justfile:60]: Merged in 0fc5e55.

**Pull request merged**